src/test/miner_tests.cpp

```diff
@@ -187,6 +187,39 @@ BOOST_AUTO_TEST_CASE(CreateNewBlock_validity)
       // printf("%lu %lu <= %lu\n", (long unsigned int) blockSize, (long unsigned int) pblocktemplate->block.nBlockSize, (long unsigned int) maxGeneratedBlock);
       delete pblocktemplate;
     }
+
+    coinbaseReserve.value=0;  // Test no reserve
+     // Now generate lots of full size blocks and verify that none exceed the maxGeneratedBlock value
+    for (unsigned int i = 2000; i <= 80000; i+=37)
+    {
+      maxGeneratedBlock = i;
+
+      pblocktemplate = CreateNewBlock(chainparams, scriptPubKey);
+      BOOST_CHECK(pblocktemplate);
+      BOOST_CHECK(pblocktemplate->block.fExcessive == false);
+      BOOST_CHECK(pblocktemplate->block.nBlockSize <= maxGeneratedBlock);
+      unsigned int blockSize = pblocktemplate->block.GetSerializeSize(SER_NETWORK, CTransaction::CURRENT_VERSION);
+      BOOST_CHECK(blockSize <= maxGeneratedBlock);
+      //printf("%lu %lu <= %lu\n", (long unsigned int) blockSize, (long unsigned int) pblocktemplate->block.nBlockSize, (long unsigned int) maxGeneratedBlock);
+      delete pblocktemplate;
+    }
+
+    std::string testMinerComment("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLM");
+     // Now generate lots of full size blocks and verify that none exceed the maxGeneratedBlock value
+    //printf("test mining with different sized miner comments");
```

Comment:
@gandrewstone 
1) I think that by the time you get to running the second loop of tests there will be no more transactions in the pool since they would have been mined out by then, so you'll just be mining empty blocks. (2)  I think you should put a step in there that modifies the miner string "after" the block is created and before we submit the block the way the miners do it...and then we submit the block and check that it was still accepted to and updates the blockchain.